astroquery/heasarc/core.py

```diff
@@ -43,7 +48,7 @@ def query_mission_list(self, cache=True, get_query_payload=False):
         Returns a list of all available mission tables with descriptions
         """
         request_payload = self._args_to_payload(
-            Entry='none',
+            entry='none',
```

Comment:
is this definitely right?  `Entry` is used below, but you changed to `entry` here.  It might be better to use the same capitalization everywhere for consistency; maybe it's better to change all to `Entry`, since that appears to be what the service expects?

Reply:
The `_args_to_payload` function actually documents `entry`:

https://github.com/astropy/astroquery/blob/554d35fee2f1339b90a75862944c968c8eda61e4/astroquery/heasarc/core.py#L241-L245

and also actually uses it:

https://github.com/astropy/astroquery/blob/554d35fee2f1339b90a75862944c968c8eda61e4/astroquery/heasarc/core.py#L283-L286

Passing `Entry` was silently ignored since there was no check for unused parameters. I guess it was just a typo, and the developers anticipated to use `entry`.
It had no impact though, since in this particular call it is set to `none` which is the default when it is not set at all.

Note in the quote above that `action` function argument is similarly transformed to `Action` for the service.
This way the function arguments are always lowercase, and are transformed to different cases as expected by the service, which depends on the argument.

I could certainly change it so that Entry and Action are always capitalized.
Then I would probably also change other parameters, and descriptions.
Things like `radius`:

https://github.com/astropy/astroquery/blob/554d35fee2f1339b90a75862944c968c8eda61e4/astroquery/heasarc/core.py#L325-L327

But what worries me is that this will change user interface:

https://github.com/astropy/astroquery/blob/554d35fee2f1339b90a75862944c968c8eda61e4/astroquery/heasarc/tests/test_heasarc_remote.py#L68

Which is something I was specifically trying to avoid. Would not want to break somebody's (and our own) scripts.

So let me know what you prefer! 
Should I change all arguments, including those provided to the user, to match those expected by the service?
Only change some?

Reply:
You raise great points here.

OK, let's stick with the lowercase for user-facing parameters, and therefore a mix of cases throughout the code (unfortunately).  It may be useful to include a comment briefly summarizing why we're using mixed case - i.e., just summerize what you said above, but only in 1 line.